server/master/src/main/java/org/apache/accumulo/master/metrics/Metrics2ReplicationMetrics.java

```diff
@@ -58,6 +59,7 @@ protected void snapshot() {
     registry.add(PENDING_FILES, getNumFilesPendingReplication());
     registry.add(NUM_PEERS, getNumConfiguredPeers());
     registry.add(MAX_REPLICATION_THREADS, getMaxReplicationThreads());
+    registry.add(LATENCY, getLatencyInSeconds());
```

Comment:
You're collecting the "maximum" latency in the system which isn't ideal as it paints a very skewed picture of the actual system.

For example, if you have a single file which cannot be replicated (say, it's corrupted) it would have a very long latency. The rest of the files may be replicating in a (hypothetically) 5seconds. The maximum latency would continue to increase without bound, indicating to an operator that the system is not healthy, when the system is actually 99% healthy.

Hadoop metrics2 exposes a histogram class for metrics which automatically aggregates min, median, avg, max as well as percentiles (e.g. p75, p85, p95, p98, p99).